tensorflow/python/saved_model/load.py

```diff
@@ -659,13 +660,30 @@ def uninitialized_variable_creator(next_creator, **kwargs):
     with ops.get_default_graph()._variable_creator_scope(  # pylint: disable=protected-access
         uninitialized_variable_creator,
         priority=50):
-      return variables.Variable(
-          shape=proto.shape,
-          dtype=proto.dtype,
-          name=name,
-          trainable=trainable,
-          synchronization=synchronization,
-          aggregation=aggregation), setattr
+      saved_device = proto.device
+      load_with_device = (
+          self._save_options.experimental_variable_policy
+              ._save_variable_devices()
+          and config.get_soft_device_placement()
```

Comment:
Why does soft placement matter here?

Reply:
To make the behavior to be consistent if config.get_soft_device_placement() is disabled.  An error is raised when an Op cannot be placed onto its intended device. So just to avoid sneakily fallbacking to CPU.

Reply:
Makes sense